infra/cifuzz/fuzz_target.py

```diff
@@ -111,68 +113,57 @@ def _download_corpus(self):
+
+  def fuzz(self):
     """Starts the fuzz target run for the length of time specified by duration.
 
     Returns:
       FuzzResult namedtuple with stacktrace and testcase if applicable.
     """
     logging.info('Running fuzzer: %s.', self.target_name)
-    if extra_libfuzzer_options is None:
-      extra_libfuzzer_options = []
-    env = base_runner_utils.get_env(self.config, self.workspace)
-    # TODO(metzman): Is this needed?
-    env['RUN_FUZZER_MODE'] = 'interactive'
-
-    if use_corpus:
-      # If corpus can be downloaded, use it for fuzzing.
-      self._download_corpus()
-      env['CORPUS_DIR'] = self.latest_corpus_path
-
-    options = LIBFUZZER_OPTIONS.copy() + [
-        f'-max_total_time={self.duration}',
-        # Make sure libFuzzer artifact files don't pollute $OUT.
-        f'-artifact_prefix={self.workspace.artifacts}/'
-    ] + extra_libfuzzer_options
-    command = ['run_fuzzer', self.target_name] + options
-
-    logging.info('Running command: %s', command)
-    process = subprocess.Popen(command,
-                               env=env,
-                               stdout=subprocess.PIPE,
-                               stderr=subprocess.PIPE)
 
-    try:
-      _, stderr = process.communicate(timeout=self.duration + BUFFER_TIME)
-    except subprocess.TimeoutExpired:
-      logging.error('Fuzzer %s timed out, ending fuzzing.', self.target_name)
-      return FuzzResult(None, None, self.latest_corpus_path)
+    self._download_corpus()
+    corpus_path = self.latest_corpus_path
+
+    logging.info('Starting fuzzing')
+    with clusterfuzz.environment.Environment(config_utils.DEFAULT_ENGINE,
+                                             self.config.sanitizer,
+                                             self.target_path,
+                                             interactive=True) as env:
+      engine_impl = clusterfuzz.fuzz.get_engine(config_utils.DEFAULT_ENGINE)
+      options = engine_impl.prepare(corpus_path, env.target_path, env.build_dir)
+      options.merge_back_new_testcases = False
+      options.analyze_dictionary = False
+
+      result = engine_impl.fuzz(self.target_path, options,
+                                self.workspace.artifacts, self.duration)
 
     # Libfuzzer timeout was reached.
-    if not process.returncode:
+    if not result.crashes:
       logging.info('Fuzzer %s finished with no crashes discovered.',
                    self.target_name)
       return FuzzResult(None, None, self.latest_corpus_path)
 
-    # Crash was discovered.
-    logging.info('Fuzzer %s, ended before timeout.', self.target_name)
-    testcase = get_testcase(stderr)
-    if not testcase:
-      logging.error(b'No testcase found in stacktrace: %s.', stderr)
-      return FuzzResult(None, None, self.latest_corpus_path)
+    # Only report first crash.
+    crash = result.crashes[0]
+    logging.info('Fuzzer: %s. Detected bug:\n%s', self.target_name,
```

Comment:
Will this work? Is crash.stacktrace a non-binary string?

Reply:
Yep, libClusterFuzz returns non-binary strings.